### DISCUSSION TITLE
How to override vectorization cost-model?

### DISCUSSION THREAD

**Discussion by HariAbram** [pocl/pocl#1672] [General]
I have an Intel-LLVM SYCL compiler running with the POCL backend. I am observing that kernels are not vectorizing, when I look at compilation flow information, it says the cost model is preventing vectorization. Is there a way to override the POCL vector cost model? I want to analyze what type of vector instructions POCL can generate.

**Comment by pjaaskel** [comment]
PoCL delegates the final vectorization decisions to the upstream LLVM loop vectorizer. This way it can utilize the cost model of it without implementing its own downstream.
